--- os_defines.h	Sun Apr 01 02:50:10 2012
+++ os_defines.h	Fri Aug 24 20:42:27 2012
@@ -57,9 +57,6 @@
 // See libstdc++/20806.
 #define _GLIBCXX_HAVE_DOS_BASED_FILESYSTEM 1
 
-// See  libstdc++/37522.
-#define _GLIBCXX_HAVE_BROKEN_VSWPRINTF 1
-
 // See libstdc++/43738
 // On native windows targets there is no ioctl function. And the existing
 // ioctlsocket function doesn't work for normal file-descriptors.
